boards/common/silabs/drivers/aem/aem.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2018 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_common_silabs_drivers_aem
+ * @{
+ *
+ * @file
+ * @brief       Implementations of the advanced energy monitor driver.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#include "aem.h"
+
+#include "em_dbg.h"
+#include "em_gpio.h"
+
+void aem_init(void)
+{
+    /* do not initialize if debugger is not connected */
+    if (!DBG_Connected()) {
+        return;
+    }
+
+    /* enable GPIO clock for configuring SWO pins */
+    CMU_ClockEnable(cmuClock_HFPER, true);
+    CMU_ClockEnable(cmuClock_GPIO, true);
+
+    /* enable debug peripheral via SWO */
+#ifdef _SILICON_LABS_32B_SERIES_0
+    DBG_SWOEnable(GPIO_ROUTE_SWLOCATION_LOC0);
+#else
+    DBG_SWOEnable(GPIO_ROUTELOC0_SWVLOC_LOC0);
+#endif
+
+    /* enable trace in core debug */
+    CoreDebug->DHCSR |= CoreDebug_DHCSR_C_DEBUGEN_Msk;
+    CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
+
+    /* enable PC and IRQ sampling output */
+    DWT->CTRL = 0x400113FF;
+
+    /* set TPIU prescaler to 16 */
+    TPI->ACPR = 15;
+
+    /* set protocol to NRZ */
+    TPI->SPPR = 2;
+
+    /* disable continuous formatting */
+    TPI->FFCR = 0x100;
+
+    /* unlock ITM and output data */
+    ITM->LAR = 0xC5ACCE55;
+    ITM->TCR = 0x10009;
```

Comment:
Here and above, I don't remember if there was already a reason to have this "magic numbers" (maybe they're not magic, jut I don't know the context) directly as numbers and not using `#define` ?

Reply:
TBH I don't even know what these values do :-) It is copy-paste from the examples provided by Silicon Labs.